core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -919,6 +929,26 @@ private[spark] class SparkSubmit extends Logging {
         log"disabled")
     }
 
+    if (clusterManager == EXTERNAL) {
+      sparkConf.get(EXTERNAL_CLUSTER_SUBMIT_CLASS) match {
+        case Some(submitClass) if !Utils.classIsLoadable(submitClass) && !Utils.isTesting =>
+          error(
+            s"Could not load external cluster submit class: $submitClass. " +
+              "Make sure the classes are included in the .")
+        case Some(_) =>
+        case None =>
+          error(
+            "External cluster submit class must be specified when running in external cluster mode."
+          )
```

Comment:
I'd turn this into explicit if clauses, then your expectations become more readable:

```suggestion
      val maybeSubmitClass = sparkConf.get(EXTERNAL_CLUSTER_SUBMIT_CLASS)
      if (maybeSubmitClass.isEmpty()) {
          error(
            "External cluster submit class must be specified when running in external cluster mode."
          )
      }
      maybeSubmitClass.forEach { submitClass =>
          if (!Utils.classIsLoadable(submitClass) && !Utils.isTesting) {
            error(
              s"Could not load external cluster submit class: $submitClass. " +
                "Make sure the classes are included in the .")
          }
        }
```